client/js/color.js

```diff
@@ -0,0 +1,42 @@
+export function toRGB(inputColor) {
+  let hex = toHex(inputColor)
+  hex = hex.slice(1);
+  var rgb = hex.match(new RegExp('(.{' + hex.length/3 + '})', 'g')).map(function(l) {
+    return parseInt(hex.length%2 ? l+l : l, 16);
+  });
+  return `rgb(${rgb.join(',')})`;    
+}
+
+export function contrastAnyColor(inputColor, intensity, direction) {
+  // Code created by https://chat.openai.com/chat
+  let color = toHex(inputColor);
+  const r = parseInt(color.slice(1, 3), 16);
+  const g = parseInt(color.slice(3, 5), 16);
+  const b = parseInt(color.slice(5, 7), 16);
+  const luminance = (0.2126 * r + 0.7152 * g + 0.0722 * b) / 255;
```

Comment:
Might be nice to move luminance calculation to it's own function. Would be nice to have a function that calculates the [contrast between colors](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#contrast-ratiodef) which would also make use of the luminance calculation.

Reply:
Done